src/_pytest/pathlib.py

```diff
@@ -365,3 +365,105 @@ def fnmatch_ex(pattern: str, path) -> bool:
 def parts(s: str) -> Set[str]:
     parts = s.split(sep)
     return {sep.join(parts[: i + 1]) or sep for i in range(len(parts))}
+
+
+def import_module(path, modname=None, ensuresyspath=True):
+    """ return path as an imported python module.
+
+    If modname is None, look for the containing package
+    and construct an according module name.
+    The module will be put/looked up in sys.modules.
+    if ensuresyspath is True then the root dir for importing
+    the file (taking __init__.py files into account) will
+    be prepended to sys.path if it isn't there already.
+    If ensuresyspath=="append" the root dir will be appended
+    if it isn't already contained in sys.path.
+    if ensuresyspath is False no modification of syspath happens.
+
+    Special value of ensuresyspath=="importlib" is intended
```

Comment:
I guess we should integrate this section into the above now, it's funny to read "purely for using in pytest" inside pytest :)